dev/TeachingTip/TeachingTip.cpp

```diff
@@ -848,14 +881,19 @@ void TeachingTip::OnActionButtonClicked(const winrt::IInspectable&, const winrt:
 void TeachingTip::OnPopupClosed(const winrt::IInspectable&, const winrt::IInspectable&)
 {
     m_popup.get().Child(nullptr);
+    m_lightDismissIndicatorPopup.get().IsOpen(false);
     auto myArgs = winrt::make_self<TeachingTipClosedEventArgs>();
+    myArgs->Reason(m_lastCloseReason);
+    m_closedEventSource(*this, *myArgs);
```

Comment:
Why doesn't this call RaiseClosingEvent instead? Seems weird that one way you get a deferral that works and the other you don't.

Reply:
This is the event handler attached to the Popup's closed event. The golden path for closing a tip is to indicate to the tip that you want it to close, either by setting TeachingTip.IsOpen to false, clicking on a close button, or through light dismiss if the property is set. Doing this causes some actions to occur, when animations are available it starts the animation and attaches to the animations completion event to close the internal popup.  Once the internal popup has actually closed, we raise the closed event in this handler.

It is technically possible for the popup to be closed without going through this path though, for instance you could tree walk to the popup and close it yourself, in this scenario, we can't raise a closing event because the popup has already closed, but we can at least raise a closed event using this handler.

After writing this explanation I realize that I still need to ensure that the IsOpen property reflects reality.

Reply:
Oh closed vs closing. Got it.